### PR TITLE
refactor(capture): add timestamp event header

### PR DESCRIPTION
**Pull request by pl**
## Problem

We want to be able to parse Kafka message bodies concurrently, but first we need to know distinct IDs. Cookieless needs timestamps to generate distinct IDs, which right now it extracts from the message body.

## Changes

- Adds a `timestamp` Kafka message header
- Computes the timestamps in capture using the current Node.js algorithm (minus edge cases)

## How did you test this code?

- Added unit tests
- Will verify headers in prod

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: rust/capture/src/v0_request.rs [hunk 1]
```diff
@@ -418,6 +419,7 @@ pub struct ProcessedEvent {
 pub struct ProcessedEventMetadata {
     pub data_type: DataType,
     pub session_id: Option<String>,
+    pub computed_timestamp: Option<chrono::DateTime<chrono::Utc>>,
 }
```
Comment (eli-r-ph) [inline rust/capture/src/v0_request.rs:423]: should be good to remove the `chrono::` prefixes since these are imported above

**Review by eli-r-ph — APPROVED** [review]
Reviewed Rust changes in detail, given sync meeting discussion about the goal here this AM, this all LGTM 👍

File: plugin-server/src/ingestion/event-preprocessing/apply-force-overflow-restrictions.ts [hunk 2]
```diff
@@ -8,20 +7,11 @@ export type ForceOverflowDecision = {
...
-
-    message.headers?.forEach((header) => {
-        if ('distinct_id' in header) {
-            distinctId = header['distinct_id'].toString()
-        }
-        if ('token' in header) {
-            token = header['token'].toString()
-        }
-    })
+    const distinctId = headers?.distinct_id
```
Comment (jose-sequeira) [inline plugin-server/src/ingestion/event-preprocessing/apply-force-overflow-restrictions.ts]: Should we have a proper handling in case headers aren't there? Maybe so rogue product is producing to the topic

File: plugin-server/src/ingestion/ingestion-consumer.ts [hunk 3]
```diff
@@ -610,46 +610,54 @@ export class IngestionConsumer {
...
             }
 
-            const parsedEvent = parseKafkaMessage(filteredMessage)
+            const parsedEvent = parseKafkaMessage(message)
             if (!parsedEvent) {
                 continue
             }
 
-            const eventWithTeam = await resolveTeam(this.hub, parsedEvent)
+            const eventWithHeaders = { ...parsedEvent, headers }
```
Comment (jose-sequeira) [inline plugin-server/src/ingestion/ingestion-consumer.ts]: ~~Could we possibly override something we shouldn't? This merge seems a little weird, namely because it seems to comply with the `IncomingEvent` interface, but, doesn't this just write `distinct_id` , `timestamp` and `token` over the values in the event?~~

Nvm I was seeing and outdated `types.ts` in my computer

File: plugin-server/src/types.ts [hunk 4]
```diff
@@ -1344,15 +1347,23 @@ export interface PipelineEvent extends Omit<PluginEvent, 'team_id'> {
     token?: string
 }
 
+export interface EventHeaders {
```
Comment (jose-sequeira) [inline plugin-server/src/types.ts]: Should they be required? Seems that they should
